bluesky_adaptive/adjudicators/base.py

```diff
@@ -0,0 +1,236 @@
+                judgment = Judgment(*judgment)  # Validate
+            self._add_suggestion_to_queue(**judgment.dict())
+
+    @abstractmethod()
+    def make_judgments(self) -> Sequence[Tuple[API_Threads_Mixin, str, Suggestion]]:
+        """Instance method to make judgements based on current suggestions.
+        The returned tuples will be deconstructed to add suggestions to the queue.
+        """
+        ...
+
+
+class AgentByNameAdjudicator(AdjudicatorBase):
+    """Adjudicator that only allows messages from a set primary agent, and uses a single qserver.
+    Parameters
+    ----------
+    qservers :  dict[str, API_Threads_Mixin]
+        Dictionary of objects to manage communication with Queue Server. These should be keyed by the beamline TLA
+        expected in AdjudicatorMsg.suggestions dictionary.
+    """
+
+    def __init__(self, *args, qservers: dict[str, API_Threads_Mixin], **kwargs):
+        self._primary_agent = ""
+        self._re_managers = qservers
+        super().__init__(*args, **kwargs)
+
+    @property
+    def primary_agent(self):
+        return self._primary_agent
+
+    @primary_agent.setter
+    def primary_agent(self, name: str):
+        self._primary_agent = "str"
+
+    def server_registrations(self) -> None:
+        self._register_property("priamry_agent")
+        super().server_registrations()
+
+    def make_judgments(self) -> Sequence[Tuple[API_Threads_Mixin, str, Suggestion]]:
+        judgments = []
+
+        if self.primary_agent not in self.agent_names:
+            logger.debug(f"Agent {self.primary_agent} not known to the Adjudicator")
+        else:
+            adjudicator_msg = self.current_suggestions[self.primary_agent]
+            for key, manager in self._re_managers.items():
+                suggestions = adjudicator_msg.suggestions.get(key, [])
+                for suggestion in suggestions:
+                    judgments.append(
+                        Judgment(re_manager=manager, agent_name=self.primary_agent, suggestion=suggestion)
+                    )
+        return judgments
+
+
+class NonredundantAdjudicator(AdjudicatorBase):
+    """Use a hashing function to convert any suggestion into a unique hash.
+
+    Parameters
+    ----------
+    topics : list of str
+        List of existing_topics as strings such as ["topic-1", "topic-2"]
+    bootstrap_servers : str
+        Comma-delimited list of Kafka server addresses as a string
+        such as ``'broker1:9092,broker2:9092,127.0.0.1:9092'``
+    group_id : str
+        Required string identifier for the consumer's Kafka Consumer group.
+    qservers :  dict[str, API_Threads_Mixin]
+        Dictionary of objects to manage communication with Queue Server. These should be keyed by the beamline TLA
+        expected in AdjudicatorMsg.suggestions dictionary.
+    hash_suggestion : Callable
+        Function that takes the tla and Suggestion object, and returns a hashable object.
```

Comment:
```suggestion
        Function that takes the tla and Suggestion object, and returns a hashable object as ::
           
            def hash_suggestion(tla: str, suggestion: Suggestion) -> Hashable: ...
            
        
```

Reply:
minor doc comment, but I could think of a handful of signatures that would be fairly described by the prose.